Docs/stablecoin/usdc usage (#21308)
MystenLabs/sui: docs/content/guides/developer/stablecoins.mdx

### Using USDC in Move

USDC uses the [Sui Coin standard](../../standards/coin.mdx) and can be used just like any other coin type in the Sui framework.

Once the `usdc` package is imported, you can use the `USDC` type.

```move
use usdc::usdc::USDC;
```

Then use the `USDC` type just as you would use the `SUI` type when accepting a `Coin<SUI>` parameter.

{@inject: examples/move/usdc_usage/sources/example.move#fun=buy_sword_with_usdc}

The following example demonstrates how to use the USDC stablecoin in a Move package and how it relates to using the `SUI` type as well as any generic coin types.

{@inject: examples/move/usdc_usage/sources/example.move}

### Using USDC in PTBs

Use USDC in your [PTBs](https://sdk.mystenlabs.com/typescript/transaction-building/basics) just like any other coin type.

Create a `Coin<USDC>` object with the `coinWithBalance` function.

```typescript
const usdcCoin = coinWithBalance({
  type: '0xa1ec7fc00a6f40db9693ad1415d0c193ad3906494428cf252621037bd7117e29::usdc::USDC',
  balance: 1_000_000
})
```

> `0xa1ec7fc00a6f40db9693ad1415d0c193ad3906494428cf252621037bd7117e29` is the [Testnet address](https://suiscan.xyz/testnet/coin/0xa1ec7fc00a6f40db9693ad1415d0c193ad3906494428cf252621037bd7117e29::usdc::USDC) for USDC, while `0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7` is the [Mainnet address](https://suiscan.xyz/coin/0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7::usdc::USDC).

Reviewer: Might want to add a `:::info:::` here mentioning that the address in the moveCall below is the address of the published package described in the USDC in Move section.

Author: Why use the `:::info:::`? do you think it's more attention grabbing?

**Site:** [Ondo Finance](https://ondo.finance/)

## How to use USDC on Sui {#usdc-guide}

:::info
While this example uses USDC, the same principles can be applied to any asset on Sui that uses the [Sui Coin standard](../../standards/coin.mdx).
:::

### Prerequisites

- Make sure you have some USDC tokens. Get Testnet tokens from Circle's [faucet](https://faucet.circle.com/).

> ### USDC Stablecoin source code

Reviewer 2: not sure if we should capitalize stablecoin. L52 doesn't so wouldn't here.

The USDC stablecoin source code is available in the [circlefin/stablecoin-sui](https://github.com/circlefin/stablecoin-sui/blob/master/packages/usdc/sources/usdc.move) repository.

### Import the USDC module in your Move package

To import the USDC module, add the following line to the `[dependencies]` section of your Move package's `Move.toml` file:

```toml
usdc = { git = "https://github.com/circlefin/stablecoin-sui.git", subdir = "packages/usdc", rev = "master" }
```

After importing the module, your Move package should look like the following:

{@inject: examples/move/usdc_usage/Move.toml}

:::warning

> The `usdc` package uses a specific version of the `sui` package. This will cause a version conflict with the `Sui` package in the Sui framework. You can override the version of the `Sui` package in your `Move.toml` file to use a different version. Add the `override = true` flag to the `Sui` package in your `Move.toml` file.

Reviewer 2: ```suggestion
The `usdc` package uses a specific version of the `sui` package, which causes a version conflict with the `Sui` package in the Sui framework. You can override the version of the `Sui` package in your `Move.toml` file to use a different version. Add the `override = true` flag to the `Sui` package in your `Move.toml` file.
```
fyi...and you might already know this but we change that local import in examples to testnet for sui dependency in the toml programmatically when we use that code import thingy.

Author: thanks for the heads up. This should be fine!

:::

### Using USDC in Move

USDC uses the [Sui Coin standard](../../standards/coin.mdx) and can be used just like any other coin type in the Sui framework.

> Once the `usdc` package is imported, you can use the `USDC` type.

Reviewer 2: ```suggestion
After importing the `usdc` package, you can use the `USDC` type.
```
